packages/rs-drive/src/drive/batch/drive_op_batch/contract.rs

```diff
@@ -78,3 +83,21 @@ impl DriveLowLevelOperationConverter for DataContractOperationType<'_> {
         }
     }
 }
+
+impl DriveOperationWithFinalizeTasks for DataContractOperationType<'_> {
+    fn finalize_tasks(
+        &self,
+        _platform_version: &PlatformVersion,
+    ) -> Option<Vec<DriveOperationFinalizeTask>> {
+        let data_contract = match self {
+            Self::ApplyContractWithSerialization { contract, .. } => contract,
+            Self::ApplyContract { contract, .. } => contract,
+        };
+
+        let tasks = vec![DriveOperationFinalizeTask::RemoveDataContractFromCache {
+            contract_id: data_contract.id(),
+        }];
+
+        Some(tasks)
+    }
```

Comment:
A few things, first we want this to be in a finalize tasks v0. We definitely need versioning here.

Next, we want the tasks to come from the operation type, not a data contract from the operation type.

Next you should use a pipe between the two applies.